### DISCUSSION TITLE
Virtual keyboard for all fheroes2 versions

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#6887] [Ideas]
I believe there could potentially exist some touch devices on windows or linux.
So, I think we should make it possible to enable virtual keyboard, which was introduced in the 1.0.2 release, at least in configuration file for all versions. 
Also, a player can have broken keyboard on his PC, so using virtual keyboard is a nice solution to the case.

**Comment by Mr-Bajs** [comment]
Sure would be nice. Tablet and Laptop seams to merge into one and another with the so called 2-in-1 devices.

Popular devices such as Microsoft Surfaces would surely benefit with virtual keyboard support. A nice solution would probaby be if fhereos2 would detect if a keyboard is plugged in, if no keyboard is detected then it enables the virtual keyboard.

**Comment by Branikolog** [comment]
Thanks our team for implementing this featre!
